edi_oca/models/edi_exchange_record.py

```diff
@@ -0,0 +1,572 @@
+
+    @api.constrains("edi_exchange_state")
+    def _constrain_edi_exchange_state(self):
+        for rec in self:
+            if rec.edi_exchange_state in ("new", "validate_error"):
+                continue
+            if not rec.edi_exchange_state.startswith(rec.direction):
+                raise exceptions.ValidationError(
+                    _("Exchange state must respect direction!")
+                )
+
+    @api.depends("related_exchange_ids.type_id")
+    def _compute_ack_exchange_id(self):
+        for rec in self:
+            rec.ack_exchange_id = rec._get_ack_record()
+
+    def _get_ack_record(self):
+        if not self.type_id.ack_type_id:
+            return None
+        return fields.first(
+            self.related_exchange_ids.filtered(
+                lambda x: x.type_id == self.type_id.ack_type_id
+            ).sorted("id", reverse=True)
+        )
+
+    def _compute_ack_expected(self):
+        for rec in self:
+            rec.ack_expected = bool(self.type_id.ack_type_id)
+
+    @api.depends("res_id", "model")
+    def _compute_related_record_exists(self):
+        for rec in self:
+            rec.related_record_exists = bool(rec.record)
+
+    def needs_ack(self):
+        return self.type_id.ack_type_id and not self.ack_exchange_id
+
+    _rollback_state_mapping = {
+        # From: to
+        "output_error_on_send": "output_pending",
+        "output_sent_and_error": "output_pending",
+        "input_receive_error": "input_pending",
+        "input_processed_error": "input_received",
+    }
+
+    @api.depends("edi_exchange_state")
+    def _compute_retryable(self):
+        for rec in self:
+            rec.retryable = rec.edi_exchange_state in self._rollback_state_mapping
+
+    @property
+    def record(self):
+        # In some case the res_model (and res_id) could be empty so we have to load
+        # data from parent
+        if not self.model and not self.parent_id:
+            return None
+        if not self.model and self.parent_id:
+            return self.parent_id.record
+        return self.env[self.model].browse(self.res_id).exists()
+
+    def _set_file_content(
+        self, output_string, encoding="utf-8", field_name="exchange_file"
+    ):
+        """Handy method to no have to convert b64 back and forth."""
+        self.ensure_one()
+        if not isinstance(output_string, bytes):
+            output_string = bytes(output_string, encoding)
+        self[field_name] = base64.b64encode(output_string)
+
+    def _get_file_content(
+        self, field_name="exchange_file", binary=True, as_bytes=False
+    ):
+        """Handy method to not have to convert b64 back and forth."""
+        self.ensure_one()
+        if not self[field_name]:
+            return ""
+        if binary:
+            res = base64.b64decode(self[field_name])
+            return res.decode() if not as_bytes else res
+        return self[field_name]
+
+    def name_get(self):
```

Comment:
you should now override `_compute_display_name` instead, adding the possible new fields in the depends of the method. See https://github.com/odoo/odoo/pull/122085 for more details.
For reference:
```
    @api.depends("identifier", "res_id", "model", "type_id")
    def _compute_display_name(self):
        for rec in self:
            rec_name = rec.identifier
            if rec.res_id and rec.model:
                rec_name = rec.record.display_name
            rec.display_name = f"[{rec.type_id.name}] {rec_name}"

```
Seems that you miss this document: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#tasks-to-do-in-the-migration 
please follow its guideline